### PR TITLE
Implement OverlayRulesetSelector component

### PR DESCRIPTION
**Pull request by EVAST9919**
Split from https://github.com/ppy/osu/pull/7385

### CODE REVIEW

File: osu.Game/Overlays/OverlayRulesetTabItem.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            set
+            {
+                if (accentColour == value)
+                    return;
+
+                accentColour = value;
+
+                updateState();
+            }
+        }
```
Comment (bdach) [inline osu.Game/Overlays/OverlayRulesetTabItem.cs]: This seems to be essentially identical to:
```csharp
private Bindable<Color4> accentColour;
public Color4 AccentColour
{
    get => accentColour.Value;
    set => accentColour.Value = value;
}
```
and then having a `BindValueChanged` in `LoadComplete`. I know you just moved it from its previous location but if we're touching this anyway then we might as well remove the reinvented wheel.

File: osu.Game/Overlays/OverlayRulesetTabItem.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        protected Action<Color4> OnStateUpdated;
```
Comment (bdach) [inline osu.Game/Overlays/OverlayRulesetTabItem.cs]: I'm not sure this should be a thing. Would rather have, say, a second bindable maybe?

**Review by bdach — COMMENTED** [review]
I think I'm mostly fine with this at its current state now.

**Comment by bdach** [comment]
Is this in a ready state? Hard to tell since there's been new changes since my last review of this (which seem fine).

**Comment by EVAST9919** [comment]
Yes, it is. I just added colour schemes support since https://github.com/ppy/osu/pull/7541 has been merged.

File: osu.Game/Overlays/OverlayRulesetTabItem.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+        protected override void OnActivated() => updateState();
+
+        protected override void OnDeactivated() => updateState();
+
+        private void updateState()
+        {
+            Text.Font = Text.Font.With(weight: Active.Value ? FontWeight.Bold : FontWeight.Medium);
+
+            currentColour.Value = IsHovered || Active.Value
```
Comment (peppy) [inline osu.Game/Overlays/OverlayRulesetTabItem.cs]: you can't write statements like this. it's impossible to read. split it out more.

File: osu.Game/Overlays/OverlayRulesetSelector.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Graphics;
+using osu.Game.Rulesets;
+using osuTK;
+using osuTK.Graphics;
+using System.Linq;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayRulesetSelector : RulesetSelector
```
Comment (peppy) [inline osu.Game/Overlays/OverlayRulesetSelector.cs]: this class is non-abstract but not being used anywhere directly except for a test. is this planned to be used somewhere in the future?

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/OverlayRulesetSelector.cs (hunk 4 above)
Comment (EVAST9919) [reply osu.Game/Overlays/OverlayRulesetSelector.cs]: Will be used in `RankingsOverlay`

**Review by peppy — APPROVED** [review]

**Pull request merged**